kube-client/src/client/client_ext.rs

```diff
@@ -184,6 +270,59 @@ impl Client {
         self.request::<K>(req).await
     }
 
+    /// Fetch a single instance of a `Resource` from a provided object reference.
+    ///
+    /// ```no_run
+    /// # use k8s_openapi::api::rbac::v1::ClusterRole;
+    /// # use k8s_openapi::api::core::v1::Service;
+    /// # use k8s_openapi::api::core::v1::ObjectReference;
+    /// # use k8s_openapi::api::core::v1::{Node, Pod};
+    /// # use kube::client::scope::Ref;
+    /// # use kube::{Resource, ResourceExt, api::GetParams};
+    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let client: kube::Client = todo!();
+    /// // cluster scoped
+    /// let cr: ClusterRole = client.fetch(&Ref::new(ObjectReference{
+    ///     api_version: Some("rbac.authorization.k8s.io/v1".to_string()),
+    ///     kind: Some("ClusterRole".to_string()),
+    ///     name: Some("cluster-admin".to_string()),
+    ///     ..Default::default()
+    /// }.into())).await?;
+    /// assert_eq!(cr.name_unchecked(), "cluster-admin");
+    /// // namespace scoped
+    /// let svc: Service = client.fetch(&Ref::new(ObjectReference{
+    ///     api_version: Some("v1".to_string()),
+    ///     kind: Some("Service".to_string()),
+    ///     name: Some("kubernetes".to_string()),
+    ///     namespace: Some("default".to_string()),
+    ///     ..Default::default()
+    /// }.into())).await?;
```

Comment:
I see in all the examples you are manually constructing the objectref, whereas if i were to use it, I would expect to go helpers such as [`Resource::object_ref`](https://docs.rs/kube/latest/kube/trait.Resource.html#method.object_ref).

do you have some kind of dynamic use case that requires you to construct these object references manually or this just a test convenience atm?

Reply:
Yeah, it’s just a convenience thing. With kopium now generating `ObjectReferences` automatically (thanks for the quick review and merge BTW!), this just translates to a raw field with reference somewhere. But this being a tool for CR generation, examples don’t provide options for “non-core” resources. I’ll update it with `object_ref`, SGTM